### DISCUSSION TITLE
Help understanding the Nextcloud guide

### DISCUSSION THREAD

**Discussion by kaibagley** [goauthentik/authentik#10377] [Q&A] [answered]
Hey guys,

Very new to Authentik so having trouble understanding a few things...

Trying to follow the [Nextcloud guide](https://docs.goauthentik.io/integrations/services/nextcloud/) on the site and immediately am stuck on the custom profile scope. I'm not sure what to do when the guide says:

> To set a quota set the "nextcloud_quota" property in the user's attributes...

and 

> To connect to an already existing Nextcloud user, set the "nextcloud_user_id" property in the user's attributes. This must be set for each individual user...

Where are the user's attributes?? Where am I meant to set the "nextcloud_user_id" and "nextcloud_quota" properties?

Thanks for any help!!!

**Comment by bastiane273** [comment]
Hi,
I did this:
https://blog.cubieserver.de/2022/complete-guide-to-nextcloud-saml-authentication-with-authentik/
Please pay attention to cut the Client Secret to 64 hex numbers.

BUT Redirect everything will work, but after all Nextcloud will not provision the user.

After that - I did this:
https://blog.cubieserver.de/2022/complete-guide-to-nextcloud-oidc-authentication-with-authentik/

Same error!

I do think the two apps (OAUTH AND OIDC) within nextcloud are broken!

**Comment by kaibagley** [comment, marked as answer]
I figured it out:
1. In the Authentik admin page, navigate to Directory > Users, select the user, click Edit
2. In the attributes section, set the variables "nextcloud_user_id" and "nextcloud_quota"  to your Nextcloud username, and the default user quota. 
```
nextcloud_quota: 150GB
nextcloud_user_id: <username of Nextcloud user to link this Authentik user to>
```
3. In the `user_oidc` Nextcloud app, set "user_id" as the user ID mapping

**Reply by bastiane273** [reply]
Did I understand right, that the nextcloud user has to be exist, before I connect it to authentik? In my case - the user will not be provisioned.

**Reply by kaibagley** [reply]
If the user exists beforehand, you need to do the above steps. If they don't Nextcloud should create a new user when you sign into it as an Authentik user automatically.

**Reply by BlinCT** [reply]
At the moment I'm trying to solve the default quota parameter for all users, but it doesn't work. Could you explain a little, in the Authentik settings user I specify this argument "nextcloud_quota:", and then I go to the OpenID Connect settings and there in Quota mapping I enter the same argument name?

**Reply by maximberezin97** [reply]
Thank you for posting this! I have been following Authentik's documentation and I was confused about why Nextcloud was creating a new user with a new UUID user_id whenever I added the Authentik provider. The missing part was that my user in Authentik did not have the `nextcloud_user_id` attribute, and therefore the property mapping had nothing to find. This should really be explicitly stated in Authentik's Nextcloud property mapping documentation [here](https://integrations.goauthentik.io/chat-communication-collaboration/nextcloud/#create-property-mapping-optional).

**Comment by lucasnasc2** [comment]
Did any of you guys figured out how to set the nextcloud id for new users the same as the authentik username? I have set the authentik username to be immutable, no worries there. i just don't know how to force nextcloud to stop creating new users with a random uuid.
In the nextcloud oAuth provider i have selected "Based on the User's username" option, but no luck.

**Reply by Moe1369** [reply]
Based on the users username is the correct option. On the nextcloud side it should be user id mapping "sub".
